certbot_route53/authenticator.py

```diff
@@ -14,112 +16,128 @@
 
 logger = logging.getLogger(__name__)
 
-TTL = 30
+INSTRUCTIONS = (
+    "To use, create an IAM user and attach the AmazonRoute53FullAccess policy, then store "
```

Comment:
> If you create a role and attach it to an EC2 instance, does boto3 automatically get the credentials from somewhere?

That's correct. It's uses the metadata service to figure the creds out.

> For AmazonRoute53FullAccess, when trying to set up an IAM role to be able to use Route53, I only see "Managed Policies," of which the closest match is AmazonRoute53FullAccess.

"Managed Policies" are bundles to make it easy to give people/machines perms. However, in a production scenario, you want to only give things the bare minimum they need to work. AWS works on an API call basis. From what I can read in the code, I think you only need `ListHostedZones`, `GetChange`, and `ChangeResourceRecordSets`. Here's a sample Terraform config (untested, just pulled in pieces from something we have in prod) illustrating it: https://gist.github.com/pkoch/82543aa3b8f51c540866baa4bab55f4a

> Ideally we could figure out something that's both best-practices-friendly and newbie-accessible enough to document as part of the plugin.

Maybe we can tell them something along the lines of "If you know AWS well, allow A, B, and C on the IAM role you'll be running this with. If you're not sure what that is and just want to get this working, just give yourself `AmazonRoute53FullAccess` permissions.". This tells them what they need, and also lets them know there's a "cop-out way" if they just want a cert.